src/network/connection_handle_factory.cpp

```diff
@@ -9,6 +9,8 @@ namespace terrier::network {
 ConnectionHandle &ConnectionHandleFactory::NewConnectionHandle(int conn_fd,
                                                                std::unique_ptr<ProtocolInterpreter> interpreter,
                                                                common::ManagedPointer<ConnectionHandlerTask> handler) {
+  common::SpinLatch::ScopedSpinLatch guard(&reusable_handles_latch_);
```

Comment:
I realize we don't expect a ton of contention here, but are we holding the latch longer than we need to here? It's just needed to protect the data structure, right? If so we can manually reduce the scope with another set of braces.